src/scheduler/job_info.c

```diff
@@ -573,8 +609,18 @@ query_jobs(status *policy, int pbs_sd, queue_info *qinfo, resource_resv **pjobs,
 
 	server_time = qinfo->server->server_time;
 
+	for (i = 0; jobattrs[i] != NULL; i++) {
```

Comment:
Why don't you put this code in a helper function and then call it for nodes as well.

Also, I don't see you freeing the list.

Reply:
I think I'll make it a static list as Arun suggested, that way we don't have to recreate it every cycle. But I do like the idea of also doing this for nodes, so I'll add similar code in query_nodes.